### PR TITLE
Screen options tab component with tests and feature switch

### LINKED ISSUE #53706: Nav Unification: Create UI for rendering a quick switcher on Calypso
As part of https://github.com/Automattic/wp-calypso/issues/51441, we need to create a UI component in Calypso that will render the new Calypso/WP switcher in the following pages:
- My Home
- Posts > All posts
- Posts > Add new
- Posts > Categories
- Posts > Tags
- Media
- Pages > All pages
- Pages > Add new
- Testimonials > All testimonials
- Testimonials > Add new
- Portfolio > All projects
- Portfolio > Add new
- Comments
- Appearance > Themes
- Appearance > Customizer
- Appearance > Widgets
- Appearance > Menus
- Appearance > Background
- Users > All users
- Tools > Import
- Tools > Export
- Settings > General
- Settings > General > Writing
- Settings > General > Discussion
- Site editor (FSE sites)

This task is about creating the UI pieces without any functionality (just the markup and styling) according to the Figma design file (JOrcjwreJTRNKKtd6HNYG1-fi-1821%3A289).

### PR DESCRIPTION
**Pull request by tjcafferkey**
#### Changes proposed in this Pull Request

* Create the Screen Options tab and the Screen Selector component in Calypso.

#### Testing instructions

* Visit Calypso.live link generated below and append `?flags=nav-unification/switcher` to the end of the URL to enable the Screen Options
* Test the component looks like the design and you are able to interact with it using mouse & keyboard as you'd expect.
* Test that it has been added to all the correct pages listed here https://github.com/Automattic/wp-calypso/issues/53706. With the exclusion of the list below.

Screenshot 2021-06-21 at 10 56 23

#### Additional information

The following views from the related issue have not had the Screen Options component added to them:
* Posts > Add new
* Pages > Add new
* Appearance > Customizer
* Appearance > Widgets
* Appearance > Menus
* Appearance > Background
* Site editor (FSE sites)

Fixes https://github.com/Automattic/wp-calypso/issues/53706

### CODE REVIEW

**Comment by mmtr** [comment]
> Test that it has been added to all the correct pages listed here #53706. 

We're finally adding it to My Home as well according to https://github.com/Automattic/wp-calypso/issues/50945#issuecomment-863130701 (pd2C0z-3q-p2#comment-115), but I forgot to update #53706. Can you please include it yo My Home as well?

**Comment by tjcafferkey** [comment]
>Can you please include it yo My Home as well?

Oops, of course. My bad was going off the list within the issue. Tunnel vision! Will push a commit now.

File: client/components/screen-options-tab/index.js
Comment (mmtr) [inline client/components/screen-options-tab/index.js:6]: TIL about `useI18n`. What's the difference between this and `useTranslate` from `i18n-calypso`?

**Review by mmtr — COMMENTED** [review]
Nice job! It looks great, and it reacts to my keyboards/mouse events as expected.

It matches most of the design, but I think the active/hover states are different:

Figma | This PR
--- | ---
Jun-18-2021 14-52-28 | Jun-18-2021 14-52-43

I think the tab should also uses the appropriate colors according to the current color scheme. Right now, it seems to use the accent color from the Default / Classic Bright themes (pink).

I also noted a small glitch on mobile viewports. On certain languages the page title can be quite long and is covered by the Screen Options tab:

Screen Shot 2021-06-18 at 14 39 51

File: client/components/screen-options-tab/index.js
Comment (tjcafferkey) [reply client/components/screen-options-tab/index.js:6]: There a plan to drop `i18n-calypso` I believe in favour of `useI18n`. https://github.com/Automattic/wp-calypso/issues/50699#issuecomment-795805854

**Comment by tjcafferkey** [comment]
@mmtr thanks for testing. I've made the changes to reflect the below comments

>It matches most of the design, but I think the active/hover states are different:

>I think the tab should also uses the appropriate colors according to the current color scheme

File: client/components/screen-options-tab/index.js
Comment (mmtr) [inline client/components/screen-options-tab/index.js]: What if you add the `nav-unification/switcher` to the test.json config file instead of checking `isRunningInJest`?

https://github.com/Automattic/wp-calypso/blob/trunk/config/test.json

File: client/components/screen-options-tab/style.scss
Comment (mmtr) [inline client/components/screen-options-tab/style.scss]: Do the Studio Blue colors work for all color schemes or are they specific to the Default / Classic Bright ones?

File: client/components/screen-options-tab/index.js
Comment (tjcafferkey) [reply client/components/screen-options-tab/index.js]: TIL. Done! Thanks.

File: client/my-sites/pages/main.jsx
Comment (mmtr) [inline client/my-sites/pages/main.jsx]: Nitpick:
```suggestion
					hasScreenOptions={ config.isEnabled( 'nav-unification/switcher' ) }
```

File: client/my-sites/themes/themes-header.jsx
Comment (mmtr) [inline client/my-sites/themes/themes-header.jsx:26]: Is the `hasScreenOptions` prop missing here?

Comment (mmtr) [reply client/my-sites/themes/themes-header.jsx:26]: Seems to be missing also here:

https://github.com/Automattic/wp-calypso/blob/64d9c606db7367681ebbecb4fa71c15d61d55f76/client/my-sites/types/main.jsx#L52-L58

File: client/components/screen-options-tab/style.scss
Comment (tjcafferkey) [reply client/components/screen-options-tab/style.scss]: Nope you're right, different focus colors for different themes. Made an update.

**Review by mmtr — APPROVED** [review]
Thanks for all these amendments, @tjcafferkey! This tests really well for me.

Only noted a minor issue. If you open the Screen Options tab with the mouse, and then use the keyboard to focus the "Classic view" item (after pressing the tab key twice), both items are highlighted:

Screen Shot 2021-06-18 at 16 58 54

However, when using the mouse for hovering the "Classic view" item, only that item gets highlighted:

Screen Shot 2021-06-18 at 17 04 32

Anyway, not a blocker for this PR. If you decide to address it before landing this PR, you don't need to ask for re-approval, feel free to merge it anyways.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6063914

Thank you @tjcafferkey for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6063914).

**Pull request merged**
